modules/openxBidAdapter.js

```diff
@@ -1,298 +1,331 @@
+  oxParams.dddid = utils._map(bids, bid => bid.transactionId).join(',');
+  oxParams.aus = utils._map(bids, bid => {
+    return utils.parseSizesInput(bid.sizes).join(',');
+  }).join('|');
+
+  let customParamsForAllBids = [];
+  let hasCustomParam = false;
+  bids.forEach(function (bid) {
+    if (bid.params.customParams) {
+      let customParamsForBid = utils._map(Object.keys(bid.params.customParams), customKey => formatCustomParms(customKey, bid.params.customParams));
+      let formattedCustomParams = window.btoa(customParamsForBid.join('&'));
+      hasCustomParam = true;
+      customParamsForAllBids.push(formattedCustomParams);
+    } else {
+      customParamsForAllBids.push('');
     }
+  });
+  if (hasCustomParam) {
+    oxParams.tps = customParamsForAllBids.join(',');
   }
 
-  function callBids(params) {
-    let isIfr;
-    const bids = params.bids || [];
-    let currentURL = (window.parent !== window) ? document.referrer : window.location.href;
-    currentURL = currentURL && encodeURIComponent(currentURL);
-    try {
-      isIfr = window.self !== window.top;
-    } catch (e) {
-      isIfr = false;
-    }
-    if (bids.length === 0) {
-      return;
+  let customFloorsForAllBids = [];
+  let hasCustomFloor = false;
+  bids.forEach(function (bid) {
+    if (bid.params.customFloor) {
+      customFloorsForAllBids.push(bid.params.customFloor * 1000);
+      hasCustomFloor = true;
+    } else {
+      customFloorsForAllBids.push(0);
     }
+  });
+  if (hasCustomFloor) {
+    oxParams.aumfs = customFloorsForAllBids.join(',');
+  }
 
-    let delDomain = bids[0].params.delDomain;
-    let bcOverride = bids[0].params.bc;
+  let url = `//${delDomain}/w/1.0/arj`;
+  return {
+    method: 'GET',
+    url: url,
+    data: oxParams,
+    payload: {'bids': bids, 'startTime': new Date()}
+  };
+}
+
+function buildOXVideoRequest(bids) {
+  return bids.map(function(bid) {
+    let url = 'http://' + bid.params.delDomain + '/v/1.0/avjp';
+    let oxVideoParams = generateVideoParameters(bid);
+    return {
+      method: 'GET',
+      url: url,
+      data: oxVideoParams,
+      payload: {'bid': bid, 'startTime': new Date()}
+    };
+  });
+}
+
+function generateVideoParameters(bid) {
+  let oxVideo = bid.params.video;
+  let oxVideoParams = {
+    auid: bid.params.unit,
+    url: oxVideo.url,
+    jsonp: 'json'
+  };
 
-    startTime = new Date(params.start);
-    if (params.timeout) {
-      timeout = params.timeout;
-    }
-    if (bids[0].params.hasOwnProperty('sendBoPixel') && typeof (bids[0].params.sendBoPixel) === 'boolean') {
-      shouldSendBoPixel = bids[0].params.sendBoPixel;
+  if (oxVideo.vtest) {
+    oxVideoParams.vtest = oxVideo.vtest;
+  }
+  if (oxVideo.be) {
+    oxVideoParams.be = oxVideo.be;
+  }
+
+  return oxVideoParams
+}
+
+function createVideoBidResponses(response, {bid, startTime}) {
+  let bidResponses = [];
+
+  if (response !== undefined && response.cache_key !== '') {
+    let bidResponse = {};
```

Comment:
How do you handle the case where there are multiple bids coming back in the bid response? It seems you dont loop through a list at all here, and only has one bidResponse at all times?

Reply:
The json response is like this: {"cache_key":"test_cache_key","pub_rev":"1000","per_colo_domain":"http://delivery-us-west-1.openx.net","ph":"7a3b9374-7986-4a41-a79d-034193518aee","pixels":"http://us-u-test.openx.net/w/1.0/pd?plm=1&ph=4c566c61-31ac-438d-bea8-f7cc314b1c11"}
we always return one bid response at all times.